cookbook/basic_rag_example/local_embeddings.py

```diff
@@ -0,0 +1,60 @@
+"""A simple script for using RAG to ask questions about recent soccer results."""
+import os
+
+import numpy as np
+import openai
+import pandas as pd
+
+from mirascope import OpenAIChat, Prompt
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+
+soccer_info = [
+    "Manchester City won the league, continuing their domestic dominance.",
+    "In La Liga, FC Barcelona emerged victorious with a sturdy defense.",
+    "Dortmund slipped up in the last matchday to hand Bayern the Bundesliga title.",
+    "For the first time since Diego Maradona's days, Napoli lifted the Serie A trophy.",
+]
+
+client = openai.OpenAI()
+embeddings_model = "text-embedding-ada-002"
+df = pd.DataFrame(columns=["texts", "embeddings"])
+for text in soccer_info:
```

Comment:
wondering if this would look better as a list comprehension:

```python
embeddings = [
    client.embeddings.create(model="text-embedding-ada-002", input=[text])
    .data[0]
    .embedding
    for text in soccer_info
]
df = pd.DataFrame({"texts": soccer_info, "embeddings": embeddings})
```

Reply:
This will also make the use of client = openai.OpenAI() more clear because you will use the client two lines later